http/handler.go

```diff
@@ -168,14 +168,29 @@ func respondError(w http.ResponseWriter, status int, err error) {
 	enc.Encode(resp)
 }
 
-func respondCommon(w http.ResponseWriter, resp *logical.Response) bool {
+func respondCommon(w http.ResponseWriter, resp *logical.Response, err error) bool {
 	if resp == nil {
 		return false
 	}
 
 	if resp.IsError() {
+		var statusCode int
+
+		switch err {
+		case logical.ErrPermissionDenied:
```

Comment:
I think we have to be careful/conscious here. Returning a permission denied could potentially verify the existence of a secret. Just like GitHub returns a 404 when you try to access a private repo that you don't have access to, I think we might need to always return a 404.